### ISSUE TITLE
GZip without TAR

### ISSUE THREAD

**Issue by richardgavel** [akordowski/Cake.Compression#14]
I was struggling with why my GZIP wasn't working as expected until I read the source code and realized you both TAR AND GZIP the files. The problem is my files are already in a TAR (It's a save of a Docker image, which is output as a TAR). Seems kinda of weird that you do both by default. Is there a way to allow GZIP without TAR? I know doing this by default would be a breaking change.

**Comment by akordowski** [comment]
@richardgavel I'm sorry you facing issues with the AddIn. I created the combined way to provide a convinent way to create a GZip archive. But looking at your case, that was not so smart. Unfortunatly I can't provide at the moment a quick solution for your issue. I have to look deeper at it .

**Comment by richardgavel** [comment]
At the moment, it was quickest to just call 7zip.exe, so I'm not blocked. I know I could have also done the C# code to GZIP it too and probably will.

**Comment by akordowski** [comment]
Glad to hear you have found a solution. But anyway, thank you for the hint and I will try to solve the issue.

**Comment by richardgavel** [comment]
One option, there technically is no overload of the GZipCompress alias that takes in a single FilePath as the input. I had to use the IEnumerable<FilePath> with one entry. You could that alias and skip the TAR if the single FilePath is a TAR file...Anytime you're specifying more than one file, by definition those files could still be TARred together.

**Comment by akordowski** [comment]
Yes, I had also the idea. But the `Uncompres()` method would be not the same, because there I can't determine if the archive contains a Tar file. But if it were a help for you I could implement the workaround.

**Comment by richardgavel** [comment]
Yeah, good point. I didn't think of the corresponding uncompress method
since all I needed was the compress step.

**Comment by maulwuff** [comment]
This behaviour drove me nuts for some time. I have the same use case as the topic starter: gzip a docker tar output.

**Comment by webbrain-one** [comment]
Hi! Thanks for the detailed report. I opened PR #34 to fix this. It adds an option to GZIP without TARing, which should work perfectly for your Docker image use case. Let me know what you think!
